data/compose-preview.yaml

```diff
@@ -0,0 +1,61 @@
+x-redash-service: &redash-service
```

Comment:
What's the difference between this and `compose.yaml`? Is there any way we could use the same configuration for both? Feels like that would be preferable

Reply:
Good point, I'll look into. :smile:

The most significant difference is the preview image using port 5001 for the webui.

The other ones (aside from the port 5678 addition) are probably all skippable or could be added to the non-preview one too.

Now that I look at it, I'm not sure what the new port 5678 mapping is for.  **Might** not be needed either.  Will find out. :smile:

```
$ diff compose.yaml compose-preview.yaml 
2c2
<   image: redash/redash:10.1.0.b50633
---
>   image: redash/redash:preview
13c13,14
<       - "5000:5000"
---
>       - "5001:5000"
>       - "5678:5678"
18a20,21
>     depends_on:
>       - server
21a25,26
>     depends_on:
>       - server
27a33,34
>     depends_on:
>       - server
30a38,43
>   worker:
>     <<: *redash-service
>     command: worker
>     environment:
>       QUEUES: "periodic,emails,default"
>       WORKERS_COUNT: 1
49,54d61
<   worker:
<     <<: *redash-service
<     command: worker
<     environment:
<       QUEUES: "periodic emails default"
<       WORKERS_COUNT: 1
```

Reply:
Heh Heh Heh

Port 5678 turns out to be used for remote debugging:

https://github.com/getredash/redash/blob/38a06c7ab9e6e43bed8b11b5f48a4199e640fdbf/redash/__init__.py#L20-L23

Yeah, we definitely don't want that mapped by default for production oriented setups.

Reply:
@wlach You were right, the additional compose file wasn't needed.  I've just pushed an update that uses the same compose file for both, doing some sed magic in the script to update the 5000-vs-5001 port number and Docker image name.

It all still seems happy, though I'll do a bunch more testing before I really feel this is suitable to merge. :smile: